src/qinfer/geometry.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+##
+# geometry.py: Computational geometry utilities.
+##
+# © 2016 Chris Ferrie (csferrie@gmail.com) and
+#        Christopher E. Granade (cgranade@gmail.com)
+#     
+# This file is a part of the Qinfer project.
+# Licensed under the AGPL version 3.
+##
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+##
+
+## FEATURES ###################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+## EXPORTS ###################################################################
+
+__all__ = [
+    'convex_distance',
+    'approximate_convex_hull'
+]
+
+## IMPORTS ###################################################################
+
+import numpy as np
+from scipy.spatial.distance import cdist
+
+from qinfer.utils import requires_optional_module
+
+try:
+    import cvxopt
+    import cvxopt.solvers
+except:
```

Comment:
I would catch on `ImportError` and let all others raise an issue. This is important because if you have code that can use `cvxopt`, and the package is installed, but something goes wrong (maybe an environment issue, or windows being stupid, etc) that it should halt.

If you want more rigorous separation you'll either need to use separate files (one for functions that use `cvxopt` and one for those that don't), or if the number of functions that require this package is low (but those functions in turn do require them) you can lazy-load the package by doing your import statement in the function itself. Keep in mind though that that will be called every time the function is called, and while it is fast, its not free.